projects/components/src/modal/modal.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable, Injector } from '@angular/core';
+import { PopoverBackdrop, PopoverFixedPositionLocation, PopoverPositionType } from '../popover/popover';
+import { PopoverService } from '../popover/popover.service';
+import { DefaultModalRef } from './default-modal-ref';
+import { ModalConfig, ModalRef, MODAL_DATA } from './modal';
+import { ModalConstructionData, ModalContainerComponent } from './modal-container.component';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ModalService {
+  private activeModal?: ModalRef<unknown>;
+
+  public constructor(private readonly popoverService: PopoverService, private readonly defaultInjector: Injector) {}
+
+  public createModal<TData, TResponse = never>(
+    config: ModalConfig<TData>,
+    injector: Injector = this.defaultInjector
+  ): ModalRef<TResponse> {
+    this.activeModal?.close();
+    const newModal = new DefaultModalRef();
+    this.activeModal = newModal;
+
+    const popover = this.popoverService.drawPopover({
```

Comment:
nit: this can be put into a separate function once we add the confirmation modal code to further improve readability.